src/devices/revolt_nc5462.c

```diff
@@ -0,0 +1,159 @@
+/** @file
+    Revolt NC5462 Energy Monitor
+
+    Copyright (C) 2023 Nicolai Hess
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+*/
+/**
+  revolt energy power meter.
+  sends on 433.92 MHz.
+
+  Pulse Width Modulation with startbit/delimiter
+  Two Modes:
+  Normal data mode:
+  105 pulses
+  first pulse sync
+  104 data pulse (11 times 8 bit data + 8 bit checksum + 8 bit unknown)
+  11 byte data:
+   +----------+----------+
+   |* data    |byte      |
+   +----------+----------+
+   |* id      |0,1       |
+   +----------+----------+
+   |* voltage |2         |
+   +----------+----------+
+   |* current |3,4       |
+   +----------+----------+
+   |*         |5         |
+   |frequency |          |
+   +----------+----------+
+   |* power   |6,7       |
+   +----------+----------+
+   |* power   |8         |
+   |factor    |          |
+   +----------+----------+
+   |* energy  |9,10      |
+   +----------+----------+
+   |* detect  |first bit |
+   |flag      |high      |
+   |          |yes/no    |
+   +----------+----------+
+
+
+
+  "Register" mode
+  (after pushing button on energy meter)
+  same 104 data pulses as in data mode, but first bit high and multiple rows of (the same)
+  data.
+
+  Pulses
```

Comment:
Review the formatting, this is markdown. Newlines don't actually break. Use list with dashes (`-`) for e.g. the items below.